realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -1225,10 +1230,12 @@ private void emitInsertListMethod(JavaWriter writer) throws IOException {
 
         writer.beginControlFlow("while (objects.hasNext())")
                 .emitStatement("object = (%s) objects.next()", qualifiedClassName);
-        writer.beginControlFlow("if(!cache.containsKey(object))");
+        writer.beginControlFlow("if (cache.containsKey(object))")
```

Comment:
I would be very surprised if this change was detectable even by the micro benchmarks. Anything below ~20% change is hidden by variance